sklearn/svm/_classes.py

```diff
@@ -499,6 +504,11 @@ def fit(self, X, y, sample_weight=None):
             sample_weight=sample_weight,
         )
         self.coef_ = self.coef_.ravel()
+        # Backward compatibility: _fit_liblinear is used both by LinearSVC/R
+        # and LogisticRegression but LogisticRegression sets a structured
+        # `n_iter_` attribute with information about the underlying OvR fits
+        # while LinearSVC/R only reports the maximum value.
+        self.n_iter_ = n_iter_.max().item()
```

Comment:
If we want to be 100% backward compatible:

```suggestion
        self.n_iter_ = n_iter_.max()
```

Because on `main`:

```python
from sklearn.svm import LinearSVC
from sklearn.datasets import make_classification

X, y = make_classification(random_state=42, n_classes=3, n_informative=4)
svc = LinearSVC().fit(X, y)

print(type(svc.n_iter_))
# <class 'numpy.int64'>
```

Reply:
If we want to keep this behaviour, we probably need an assert in the test.

Reply:
The attribute description says ``n_iter_ : int``. To me it's what was intended to be the type of n_iter_ and the fact that it was different so far should be considered as a bug, so I'd rather fix it. Maybe it needs a change log entry though.